File: 1.Data structure/2.Queue/프린터/프린터.java
import java.util.LinkedList;
import java.util.Queue;
import java.util.Arrays;
import java.util.stream.Collectors;

class Solution {
    public int solution(int[] priorities, int location) {
        Queue<Integer> q = new LinkedList<Integer>();
        Queue<Integer> p = new LinkedList<Integer>();
        for(int i = 0; i < priorities.length; i++){
            q.offer(priorities[i]);
            p.offer(i);
        }
        int max = q.stream().max(Integer::compare).orElse(-1);
        int answer = 0;
        while(!q.isEmpty()){
            int tempq = q.poll();
            int tempp = p.poll();
            if (tempq < max){
                q.offer(tempq);
                p.offer(tempp);
            }else if(tempp == location){
                return answer + 1;
            }else{
                max = q.stream().max(Integer::compare).orElse(-1);
                answer++;
            }
        }

        return 0;
    }
}

File: 1.Data structure/2.Queue/프린터/프린터.py
def solution(priorities, location):
    position = [x for x in range(len(priorities))]
    answer = []
    while 1:
        if len(priorities) <= 0:
            break
        elif max(priorities) <= priorities[0]:
            priorities.pop(0)
            answer.append(position.pop(0))
        else:
            priorities.append(priorities.pop(0))
            position.append(position.pop(0))
    return answer.index(location) + 1